gluten-data/src/main/java/io/glutenproject/init/BackendJniWrapper.java

```diff
@@ -16,13 +16,9 @@
  */
 package io.glutenproject.init;
 
-public class BackendJniWrapper extends JniInitialized {
+public class BackendJniWrapper {
```

Comment:
I am feeling that we can remove this jni wrapper then move the method to class `JniInitialized`.

Then we can rename `JniInitialized` to `NativeBackendInitialized` or something to make things clearer.

How do you think about it?

Reply:
LGTM, let's do refactor in follow up patch.

I would like remove class hierarchy of JniInitialized too, expose a initBackend method and invoke this in ContextInitializer Object scala class, like below:
```scala
class ContextInitializer {
  override def initialize() = {
    init
    // ...
  }
}

object ContextInitializer {
  val init = JniInitialized.initBackend
}

```

Reply:
> LGTM, let's do refactor in follow up patch.
> 
> I would like remove class hierarchy of JniInitialized too, expose a initBackend method and invoke this in ContextInitializer Object scala class, like below:
> 
> ```scala
> class ContextInitializer {
>   override def initialize() = {
>     init
>     // ...
>   }
> }
> 
> object ContextInitializer {
>   val init = JniInitialized.initBackend
> }
> ```
> 
> ContextInitializer's init variable will init in `<clinit>` as well as static initialization, thus only execute once.

Sounds good to me. Thanks.

Reply:
#3307